File: kotlin-multiplatform/src/commonMain/kotlin/com/ricoh360/thetableclient/service/data/values/PluginPowerStatus.kt
package com.ricoh360.thetableclient.service.data.values


/**
 * Plugin power status (Kind of action)
 * @see PluginControl
 */
enum class PluginPowerStatus(internal val ble: Byte?) {
    /**
     * Running (Start plugin)
     */
    RUNNING(0),

    /**
     * Stop
     */
    STOP(1),
    ;

    companion object {
        val keyName: String
            get() = "pluginControl"

        /**
         * Search by bluetooth value.
         *
         * @param bleData Return value of bluetooth api.
         * @return PluginPowerStatus
         */
        internal fun getFromBle(bleData: Byte): PluginPowerStatus? {
            return entries.firstOrNull { it.ble == bleData }
        }
    }
}

File: react-native/src/service/values/plugin-power-status.ts
/**
 * Plugin power status (Kind of action)
 * @see PluginControl
 */
export const PluginPowerStatusEnum = {
  /**
   * Running (Start plugin)
   */
  RUNNING: 'RUNNING',

  /**
   * Stop
   */
  STOP: 'STOP',
} as const;

/** type definition of PluginControlEnum */
export type PluginPowerStatusEnum =
  typeof PluginPowerStatusEnum[keyof typeof PluginPowerStatusEnum];
